### ISSUE TITLE
Make the Heartline Twister ratings less hateful

### ISSUE THREAD

**Issue by Gymnasiast** [OpenRCT2/OpenRCT2#14484]
While some roller coaster types are inevitably more popular, or comfortable, or _fun_ than others, and their ratings must reflect that, I think the ratings for the Heartline Twister are really bonkers. Marcel Vos asked himself if an Ultra Twister caused Chris Sawyer a lot of childhood trauma.

The original Ultra Twister was very popular. There seemed to be no complaints about things like comfort. Despite this, the Heartline Twister gets _penalised_ for having air time - the only roller coaster type to get it, seemingly for no good reason.

I think we should change the ratings to be more in line with other steel coaster - it is both more realistic and balanced. The lack of versatility will of course remain, so I doubt we will have to fear they'll suddenly take over the RCT2 landscape ;-)

**Comment by e-foley** [comment]
A lot of good discussion in #2031 and #2119 for extra context.

I think the reason for the penalty is to discourage players from making excruciatingly slow twists and reaping airtime benefits. It's easier to do that on rides with twists that don't accelerate/decelerate the car when going through the inversion. (I forget off-hand between common barrel rolls and in-line twists which have this property.)

There are also a bunch of rating system rules working against Heartline Twister coasters as it is: no train length benefits, no lateral-G benefits, and the lowest base excitement rating of any coaster type in the game.

**Comment by Fred-1044276** [comment]
Don't the vehicle objects for the heartline twister coaster (utcar, utcarr, tigrtwist) have a flag that says `"limitAirTimeBonus": true,`, which would mean the heartline twister cars won't get excitement bonuses from air time?

Also, of particular note is that the heartline twister coaster's rideBonusValue of 35 is the lowest of any roller coaster type in the game. It's even lower than many flat rides like the launched freefall (65), merry-go-round (45), and maze (40) for example.

**Comment by e-foley** [comment]
> yeah... a fix I could do to remove that airtime penalty all together is that you can place a maximum of one second of airtime for the bonus and do "If > 1; =1;else; = t

This is the spirit of what #2119 has already accomplished: the airtime bonus maxes out at 96 ticks (~2 seconds, I believe). In vanilla, the excess is actually *subtracted* from the excitement rating, which leads to underflows.

@Fred-1044276, there is still an airtime bonus, but it is capped, probably to deter players from making Heartline Twisters that crawl through inversions. Your point about the abnormally low excitement base is a good one though. 

Pure speculation, but it wouldn't surprise me if the low base was a first stab at discouraging slow inversions in a way ultimately effected by the airtime penalty, but then the base value wasn't raised again. This logic is similar to the Reverse Freefall Coaster, which also has a very low base rating due to large airtime potential.

**Comment by yechingu** [comment]
Heartline Twister already have terrible throughput as it is the single train coaster with low capacity. I am strongly sure about that improving stat bonus rule for Heartline Tiwster will not going to break anything. Stating balance issue-easy airtime- is actually an OUT OF SENSE EXCUSE, as we already have so many game breaking OP coaster layouts in this game ANYWAY.

ACTUAL Heartline Twister is not an intense ride, as it cannot ever be intense despite of fact the ride is designed by infamous TOGO. Nor it is a vomit machine in real world! Heartline Twister's ingame stat is total nonsense and why the hell we are keep seeing nonsense excuses to avoid addressing the issue?

It have EXPENSIVE build cost, LOW THROUGHPUT, with INSANE INTENSITY and NAUSEA RATE, and heck yeah, nobody is willing to address this issue. What an inexcusable shame.

Most of the small train coasters are pretty much an ancient relic or "token rides" which cannot have actual practicality. However, even these bad rides usually have some benefits, such as side-friction being super-easy-and-cheap-to-build-early-game-coaster (assuming that one knows how not to launch train off track).

Indeed, core design of the Heartline Twister is a bit not fitting into the core mechanic of this game, as Heartline Twister is fairly simple looking ride that provides minimal possibility for the lay-out. However it still cannot advocate the Heartline literally being SANCTIONED BUG. 

I suggest making Heartline's avg expected rating as 4/5/3, this will make sense as this game often lacks easy to build mid-intensity rides. For now, Heartline Twister cannot even be used as a "Fancy Thrill Ride" at all due to its abnormal expected rating. 

TL;DR
All those "excuses" concerning about "balance" is nonsense, hypocrisy, and farce. Just, darn, fix, it.
and ofc, something also has to be done about infamous mini-suspended coaster too as that ride is total disaster.  (maybe make it resistant to ride-value decay)

**Comment by Gymnasiast** [comment]
>  All those "excuses" concerning about "balance" is nonsense, hypocrisy, and farce. Just, darn, fix, it.

Cut the attitude. We don't appreciate being talking to like that **at all**.

**Comment by Dracheron** [comment]
yeah... @yechingu the way you addressed it was out of line

But yeah I feel like having a high excitement low intensity ride is kind of out of the question with the current way to build up good rides requires a very high intensity rating, whereas most hypers should only have a high intensity rating (and the heartline isn't all that much of a vomit machine to begin with either, its that backwards facing cars have a massive nausea boost and an intensity boost which is unwarranted)

 And speaking of stats - the multi-dimension coaster could benefit with its g's being modified (and thus its ratings) affected by the seat rotation, giving the equation <actual g's>=<base g's>*cos(seat rotation) (though forward/backwards g's should also be tracked in that case)

**Comment by duncanspumpkin** [comment]
When you see how the multi dim does rotation i think you will quickly realise why doing that would be a major pain.

**Comment by ocalhoun6** [comment]
> It have EXPENSIVE build cost

This part, at least, does make sense. In real life, it _would_ be a fairly expensive coaster to build.

(Personally, I'm in favor of at least moderately revamping the heartline twister's ratings -- and for giving it a block sectioned mode -- but I can understand why people are reluctant to deviate from the original game's ratings, even if they don't seem to make much sense sometimes.)



> the multi-dimension coaster could benefit with its g's being modified (and thus its ratings) affected by the seat rotation

That should probably be split off into a different issue. It's pretty unrelated to what this issue is about.

**Comment by Dracheron** [comment]
yeah, though at the same time I feel like it would also make them less intense and not as much of a vomit machine (i.e. X (I think Magic Mountain would be the only scenario at play but that one would probably be less intense since its done right))

reality is that either case for the heartline/multi-dimension would make things a bit interesting to calculate

Plus, there's thankfully a way of infinite convergence of calculating sine and cosine with just algebra, so getting said ratings calculated isn't going to be a big issue
